x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/diff/calculation/algorithms/rule_type_diff_algorithm.test.ts

```diff
@@ -0,0 +1,159 @@
+/*
```

Comment:
I think we should remove the scenarios which are not possible, since the user cannot edit the rule type, or change the type of the installed rule in any way, not even by API:

**Not possible:**
- ABA
- ABB
- ABC
**Possible:**
- AAA
- AAB
- -AA
- -AB

I think the not-possible scenarios shouldn't have test cases here cause it makes it confusing, maybe we can just add it as a comment at the top of this file. What do you think?

Reply:
A user can change between `query` and `saved_query` rule types in the UI. Not saying that's a super common use case but it can technically be done. I think that would be enough to keep those other scenarios in here and maybe just update the tests to reflect that instead of `eql`? I just used that for mocking simplicity since it'd be testing the same logic under the hood

Reply:
@dplumlee You are right about `query` and `saved_query`, I hadn't realized that. 

Then, considering that, and what Paula mentioned in https://github.com/elastic/kibana/pull/193372#discussion_r1778988979, I think we should leave the tests cases as they are and add explanations in comments, detailing that such scenarios are only possible for that particular rule change; and other rule type changes should not be possible in the app.